src/caret_analyze/infra/lttng/records_source.py

```diff
@@ -73,9 +73,23 @@ def publish_records(self) -> RecordsInterface:
             - source_timestamp
 
         """
-        inter_proc_publish = self._data.rclcpp_publish_instances
-
+        inter_proc_publish = self._data.rclcpp_publish_instances.clone()
         rcl_publish_records = self._data.rcl_publish_instances.clone()
+
+        # Get the publish_handle from rcl_publish and add it to the rclcpp_publish record,
+        # in case rclcpp_publish does not have a publisher_handle.
+        if len(rcl_publish_records) != 0:
+            inter_proc_publish = merge_sequential(
+                left_records=inter_proc_publish,
+                right_records=rcl_publish_records,
+                left_stamp_key=COLUMN_NAME.RCLCPP_INTER_PUBLISH_TIMESTAMP,
+                right_stamp_key=COLUMN_NAME.RCL_PUBLISH_TIMESTAMP,
+                join_left_key=COLUMN_NAME.MESSAGE,
+                join_right_key=COLUMN_NAME.MESSAGE,
+                columns=Columns.from_str(
+                    inter_proc_publish.columns + [COLUMN_NAME.PUBLISHER_HANDLE]).column_names,
+                how='left')
+
         rcl_publish_records.drop_columns([COLUMN_NAME.PUBLISHER_HANDLE])
         if len(rcl_publish_records) > 0:
```

Comment:
Does my suggested code work as well? If it works, I think it is better because it is more simple and easy to read.

```suggestion
        if len(rcl_publish_records) > 0:
```

Reply:
I have confirmed that the code you suggested works correctly.
Fixed in [f089a06](https://github.com/tier4/CARET_analyze/pull/302/commits/f089a066672b81542035cab6517447d9d1643523)